### ISSUE TITLE
`PandasDataset` can't handle mixed `Series` and `DataFrame` inputs.

### ISSUE THREAD

**Issue by jaheba** [awslabs/gluon-ts#2139]
From the tutorial:

> `PandasDataset` is meant to work on `pandas.DataFrame`, `pandas.Series` **and
any collection of both**.

However:

```
In [3]: s = pd.Series(np.arange(10), index=pd.period_range("2020", freq="D", periods=10))

In [4]: from gluonts.dataset.pandas import PandasDataset

In [7]: _ = PandasDataset(s)

In [8]: _ = PandasDataset(s.to_frame())

In [9]: _ = PandasDataset([s, s.to_frame()])
...
AttributeError: 'DataFrame' object has no attribute 'to_frame'
```

Using `PandasDataset([s.to_frame(), s])` works however.

**Comment by rsnirwan** [comment]
By 'any collection of both' is meant that it runs either on a collection of `pandas.DataFrame` or a collection of `pandas.Series`. Maybe a better wording would be: 

> `PandasDataset` is meant to work on `pandas.DataFrame`, `pandas.Series` or a collection of either one.

**Comment by jaheba** [comment]
Would it complicate things to support a mix of both? But maybe that's just too weird of use-case.

**Comment by rsnirwan** [comment]
No, it isn't much work. But I would propose not to do it anyway.. Imposing same structure on all elements of the collection seems to be nicer. If someone comes with a real world use-case for this, we can do it later.
